Bio/SeqIO/FastaIO.py

```diff
@@ -41,37 +41,28 @@ def SimpleFastaParser(handle):
 
     """
     # Skip any text before the first record (e.g. blank lines, comments)
-    while True:
-        line = handle.readline()
-        if line == "":
-            return  # Premature end of file, or just empty?
-        if line[0] == ">":
+    # This matches the previous implementation where .readline() was used
+    for line in handle:
+        if line[0] == '>':
+            title = line[1:].rstrip()
             break
 
-    while True:
-        if line[0] != ">":
-            raise ValueError(
-                "Records in Fasta files should start with '>' character")
-        title = line[1:].rstrip()
-        lines = []
-        line = handle.readline()
-        while True:
-            if not line:
-                break
-            if line[0] == ">":
-                break
-            lines.append(line.rstrip())
-            line = handle.readline()
-
-        # Remove trailing whitespace, and any internal spaces
-        # (and any embedded \r which are possible in mangled files
-        # when not opened in universal read lines mode)
-        yield title, "".join(lines).replace(" ", "").replace("\r", "")
-
-        if not line:
-            return  # StopIteration
+    # Main logic
+    # Note, remove trailing whitespace, and any internal spaces
+    # (and any embedded \r which are possible in mangled files
+    # when not opened in universal read lines mode)
+    lines = []
+    for line in handle:
+        if line[0] == '>':
+            if lines:
+                yield title, ''.join(lines).replace(" ", "").replace("\r", "")
+                lines = []
+            title = line[1:].rstrip()
+            continue
+        lines.append(line.rstrip())
 
-    assert False, "Should not reach this line"
+    if lines:
```

Comment:
I think this if statement should not be conditional - consider empty sequences (zero length), e.g.

```
>empty with a blank line afterwards

```

and:

```
>empty with no line afterwards
```

These are both valid but corner case FASTA files.

Reply:
Thanks, I did't know, is this tested in the suite? I'm not sure why the tests are failing?